src/reader.cpp

```diff
@@ -3251,6 +3309,41 @@ const wchar_t *reader_readline(int nchars) {
     return finished ? data->command_line.text.c_str() : NULL;
 }
 
+bool jump(jump_direction_t dir, jump_precision_t precision, editable_line_t *el, wchar_t target) {
+    size_t pos_start, step;
+
+    data->last_jump_target = target;
+    data->last_jump_direction = dir;
+    data->last_jump_precision = precision;
+
+    switch (dir) {
+        case jump_direction_t::forward: {
+            pos_start = el->position + 1;
+            step = 1;
+            break;
+        }
+        case jump_direction_t::backward: {
+            pos_start = el->position ? el->position - 1 : el->position;
+            step = -1;
+            break;
+        }
+    }
+
+    for (size_t tmp_pos=pos_start; tmp_pos && tmp_pos < el->size(); tmp_pos+=step) {
+        if (el->at(tmp_pos) == target) {
+            if (precision == jump_precision_t::till) {
+                tmp_pos -= step;
+            }
+            tmp_pos = std::max(tmp_pos, (size_t)0);
```

Comment:
C/C++ rust is showing

Note to self: `size_t` is unsigned, so this doesn't make sense.  I need to rework this loop else it will skip the first position in `el`.  Perhaps the `while (tmp_pos--)` loop that was there is the cleanest approach for scanning backward.